File: Year 2/Semester 2/MPP/JavaSocket/AirlineModel/src/main/java/airline/model/Flight.java
package airline.model;

import java.io.Serializable;

public class Flight extends Entity<Integer> implements Serializable {
    private final String destination;
    private String date;
    private String hour;
    private int numberOfSeats;
    private int availableSeats;

    public Flight(String destination, String date, String hour, int numberOfSeats) {
        this.destination = destination;
        this.date = date;
        this.hour = hour;
        this.numberOfSeats = numberOfSeats;
        this.availableSeats=numberOfSeats;
    }

    public int getAvailableSeats() {
        return availableSeats;
    }

    public void setAvailableSeats(int availableSeats) {
        this.availableSeats = availableSeats;
    }

    @Override
    public String toString() {
        return "Flight{" +
                "destination='" + destination + '\'' +
                ", date='" + date + '\'' +
                ", hour='" + hour + '\'' +
                ", numberOfSeats=" + numberOfSeats +
                '}';
    }

    public String getDestination() {
        return destination;
    }

    public String getDate() {
        return date;
    }

    public String getHour() {
        return hour;
    }

    public int getNumberOfSeats() {
        return numberOfSeats;
    }
}

File: Year 2/Semester 2/MPP/Protobuf/ProtoCSharp/AirlineServer/Model/Flight.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Model
{
    [Serializable]  
    public class Flight : Entity<int>
    {
        private string destination;
        private string date;
        private string hour;
        private int numberOfSeats;
        private int availableSeats;

        public Flight(string destination, string date, string hour, int numberOfSeats, int availableSeats)
        {
            this.destination = destination;
            this.date = date;
            this.hour = hour;
            this.numberOfSeats = numberOfSeats;
            this.availableSeats = availableSeats;
        }
        
        
        public string Destination { get { return destination; } }
        public string Date { get { return date; } }
        public string Hour { get { return hour; } }
        public int NumberOfSeats { get { return numberOfSeats; } }

        public int AvailableSeats
        {
            get { return availableSeats; }
            set { availableSeats = value; }
        }
        
        public override string ToString()
        {
            return "Flight{" +
                    "destination='" + destination + '\'' +
                    ", date='" + date + '\'' +
                    ", hour='" + hour + '\'' +
                    ", numberOfSeats=" + numberOfSeats +
                    '}';
        }

    }
}
